src/libs/actions/Transaction.ts

```diff
@@ -270,6 +284,99 @@ function updateWaypoints(transactionID: string, waypoints: WaypointCollection, i
     });
 }
 
+/**
+ * Dismisses the duplicate transaction violation for the provided transactionIDs
+ * and updates the transaction to include the dismissed violation in the comment.
+ */
+function dismissDuplicateTransactionViolation(transactionIDs: string[], dissmissedPersonalDetails: PersonalDetails | CurrentUserPersonalDetails) {
+    const currentTransactionViolations = transactionIDs.map((id) => ({transactionID: id, violations: allTransactionViolation?.[id] ?? []}));
+    const currentTransactions = transactionIDs.map((id) => allTransactions?.[id]);
+    const transactionsReportActions = currentTransactions.map((transaction) => ReportActionsUtils.getIOUActionForReportID(transaction.reportID ?? '', transaction.transactionID ?? ''));
+    const optimisticDissmidedViolationReportActions = transactionsReportActions.map(() =>
+        buildOptimisticDismissedViolationReportAction({reason: 'manual', violationName: CONST.VIOLATIONS.DUPLICATED_TRANSACTION}),
+    );
+    const optimisticData: OnyxUpdate[] = [];
+    const failureData: OnyxUpdate[] = [];
+
+    const optimisticReportActions: OnyxUpdate[] = transactionsReportActions.map((action, index) => ({
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${action?.childReportID ?? '-1'}`,
+        value: {
+            [optimisticDissmidedViolationReportActions[index].reportActionID]: optimisticDissmidedViolationReportActions[index] as ReportAction,
+        },
+    }));
+    const optimisticDataTransactionViolations: OnyxUpdate[] = currentTransactionViolations.map((transactionViolations) => ({
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionViolations.transactionID}`,
+        value: transactionViolations.violations?.filter((violation) => violation.name !== CONST.VIOLATIONS.DUPLICATED_TRANSACTION),
+    }));
+
+    optimisticData.push(...optimisticDataTransactionViolations);
+    optimisticData.push(...optimisticReportActions);
+
+    const optimisticDataTransactions: OnyxUpdate[] = currentTransactions.map((transaction) => ({
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.TRANSACTION}${transaction.transactionID}`,
+        value: {
+            ...transaction,
+            comment: {
+                ...transaction.comment,
+                dismissedViolations: {
+                    duplicatedTransaction: {
+                        [dissmissedPersonalDetails.login ?? '']: dissmissedPersonalDetails.accountID,
```

Comment:
Oh, tricky. So, the second argument is not the accountID but the timestamp (in seconds) of when it was dismissed. Can we change it? It will get overridden when the API call finishes anyways, but it's not used for anything meaningful.